packages/bloc/lib/src/bloc.dart

```diff
@@ -174,7 +179,8 @@ abstract class Bloc<Event, State> extends Stream<State> implements Sink<Event> {
         try {
           BlocSupervisor.delegate.onTransition(this, transition);
           onTransition(transition);
-          _stateSubject.add(nextState);
+          _stateController.add(nextState);
+          _currentState = nextState;
```

Comment:
can we also set the state before adding to the state controller?

Reply:
I am not sure what it can affect something, but was changed :)